python/lsst/pipe/drivers/background.py

```diff
@@ -0,0 +1,694 @@
+        self.config = config
+        self.dims = dims
+        self.transform = transform
+
+        if values is None:
+            values = afwImage.ImageF(self.dims)
+            values.set(0.0)
+        else:
+            values = values.clone()
+        assert(values.getDimensions() == self.dims)
+        self._values = values
+        if numbers is None:
+            numbers = afwImage.ImageF(self.dims)  # float for dynamic range and convenience
+            numbers.set(0.0)
+        else:
+            numbers = numbers.clone()
+        assert(numbers.getDimensions() == self.dims)
+        self._numbers = numbers
+
+    def __reduce__(self):
+        return self.__class__, (self.config, self.dims, self.transform, self._values, self._numbers)
+
+    def clone(self):
+        return self.__class__(self.config, self.dims, self.transform, self._values, self._numbers)
+
+    def addCcd(self, exposure):
+        """Add CCD to model
+
+        We measure the background on the CCD, and record the results
+        in the model.  For simplicity, measurements are made in a box
+        on the CCD corresponding to the warped coordinates of the
+        superpixel rather than accounting for little rotations, etc.
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            CCD exposure to measure
+        """
+        detector = exposure.getDetector()
+        transform = detector.getTransformMap().getTransform(detector.makeCameraSys(afwCameraGeom.PIXELS),
+                                                            detector.makeCameraSys(afwCameraGeom.FOCAL_PLANE))
+
+        # Bin the exposure
+        image = exposure.getMaskedImage()
+        maskVal = image.getMask().getPlaneBitMask(self.config.mask)
+
+        # Warp the binned image to the focal plane
+        toSample = transform.then(self.transform)  # CCD pixels --> focal plane --> sample
+
+        warped = afwImage.ImageF(self._values.getBBox())
+        warpedCounts = afwImage.ImageF(self._numbers.getBBox())
+        width, height = warped.getDimensions()
+
+        stats = afwMath.StatisticsControl()
+        stats.setAndMask(maskVal)
+        stats.setNanSafe(True)
+        # Iterating over individual pixels in python is usually bad because it's slow, but there aren't many.
+        pixels = itertools.product(range(width), range(height))
+        for xx, yy in pixels:
+            llc = toSample.applyInverse(afwGeom.Point2D(xx - 0.5, yy - 0.5))
+            urc = toSample.applyInverse(afwGeom.Point2D(xx + 0.5, yy + 0.5))
+            bbox = afwGeom.Box2I(afwGeom.Point2I(llc), afwGeom.Point2I(urc))
+            bbox.clip(image.getBBox())
+            if bbox.isEmpty():
+                continue
+            subImage = image.Factory(image, bbox)
+            result = afwMath.makeStatistics(subImage, afwMath.MEANCLIP | afwMath.NPOINT, stats)
+            mean = result.getValue(afwMath.MEANCLIP)
+            num = result.getValue(afwMath.NPOINT)
```

Comment:
I'm not sure I follow why multiplying by the number of pixels in the box helps. Maybe you meant "divide"? That division is done in the `getStatsImage` method.
The whole idea of keeping track of the number of pixels going into a superpixel is to avoid spurious high or low superpixels (which can easily happen near the edge because there aren't a lot of pixels there).